specs/HostResourceMapping.md

```diff
@@ -0,0 +1,103 @@
+# Background
+We have received feedbacks that there is a need to access local files in WebView2. In this document we describe the APIs for this scenario. We'd appreciate your feedback.
+
+
+# Description
+To access local files in WebView2, define a virutal host name to local folder mapping using the `AddVirtualHostNameToFolderMapping` API,
+and then refer to local files under the folder with a normal http/https url with the virtual host name.
+
+# Examples
+Suppose the app has index.html file next to its exe file. The following code snippet demonstrates how to define a mapping
+for `app-file.invalid` host name so that it can be accessed from pages using normal http/https url like https://app-file.invalid/index.html.
+
+## Win32 C++
+```cpp
+webview2->AddVirtualHostNameToFolderMapping(
+    L"app-file.invalid", L".", COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND_DENY_CORS);
+webview2->Navigate("https://app-file.invalid/index.html");
+```
+
+## .Net and WinRT
+```c#
+webView.CoreWebView2.AddVirtualHostNameToFolderMapping(
+    "app-file.invalid", ".", CoreWebView2HostResourceAccessKind.DenyCors);
+webView.Navigate("https://app-file.invalid/index.html");
+```
+
+# Remarks
+Choose a virtual host name that will not be used by real sites.  
+If you own a domain (like example.com), an option is to use a sub domain reserved for the app (like my-app.example.com).  
+If you don't own a domain, [RFC 2606](https://tools.ietf.org/html/rfc2606#section-2) has reserved several top level domain that would
+not be used by real sites, like .example, .test, and .invalid.
+
+Give only minimal cross orign access neccessary to run the app. If there is no need to access local resources
+from other sites, use COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND_DENY.
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+
+```IDL
+/// Kind of cross origin resource access allowed for host resources during download.
+/// Note that other normal access checks like same origin DOM access check and [Content
+/// Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) still apply.
```

Comment:
It would be nice to include a table here like:

| Cross-origin resource context | Deny | Allow | DenyCors |
| :------------- | :----------: | :----------: | :-----------: |
|  Image and script | Deny | Allow | Allow |
| Others | Deny | Allow | Deny |